client/components/quizSelection/TopicSelection.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { TopicSelectionContainer, TopicSelectionList, TopicSelectionItem } from './styles';
+import { PrimaryButton, PrimaryButtonSmall, TextBodySmall } from '../shared/styles';
+
+export default function TopicSelection({ subject, setChosenSubject, chosenTopics, setChosenTopics }) {
+
+  const toggleTopics = (topic) => {
+    if (chosenTopics.includes(topic)) {
+      setChosenTopics(topics => topics.filter(t => t !== topic));
+    } else {
+      setChosenTopics(topics => [...topics, topic]);
+    }
+  };
+
+  return (
+    <TopicSelectionContainer>
+      <TopicSelectionList>
+        <TopicSelectionItem>
+          <PrimaryButton mod={subject === "Programming" ? "fillLight" : "ghost"} onClick={() => {setChosenSubject("Programming"); setChosenTopics([])}}>
```

Comment:
This way, we end up having a lot of JSX if we choose to add topics.

We could add subjects as an array of objects with id and name, and loop through that to create buttons.
Same goes for topicSelection :) 

This data should probably come from backend - at that point we're forced to loop through the data. If we add it now, it should be an easy adjustment later.

Reply:
The thought was that subjects would be very limited,running a map when it results in the same 3 or 4 things seemed unnecessary to me.

Reply:
True I guess.
I do support the "rule of 3's" though, so if you need copy more than 3 times, it should be a function. Maybe that applies for sub-categories

Reply:
I misunderstood the initial ask, I thought you were asking only about the Subjects. Either way, I agree anytime we an loop over vales and save some space we should.

This was fixed in the last push, I have an array feeding the subjects/topics setup like we were discussing with Andrew, then filtering / looping for the Subjects and Topics to display.